parts/linux/cloud-init/artifacts/aks-check-network.sh

```diff
@@ -0,0 +1,206 @@
+#! /bin/bash
+# 
+# AKS Check Network
+#
+# This script is used to check network connectivity from the node to certain required AKS endpoints 
+# and log the results to the events directory. For now, this script has to be triggered manually to
+# collect the log. In the future, we will run it periodically to check and alert any issue.
+
+APISERVER_FQDN=${1:-''}
+CUSTOM_ENDPOINT=${2:-''}
+
+EVENTS_LOGGING_PATH="/var/log/azure/Microsoft.Azure.Extensions.CustomScript/events/"
+AZURE_CONFIG_PATH="/etc/kubernetes/azure.json"
+AKS_CA_CERT_PATH="/etc/kubernetes/certs/apiserver.crt"
+AKS_CERT_PATH="/etc/kubernetes/certs/client.crt"
+AKS_KEY_PATH="/etc/kubernetes/certs/client.key"
+RESOLV_CONFIG_PATH="/etc/resolv.conf"
+SYSTEMD_RESOLV_CONFIG_PATH="/run/systemd/resolve/resolv.conf"
+
+ARM_ENDPOINT="management.azure.com"
+METADATA_ENDPOINT="http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https://${ARM_ENDPOINT}/"
+API_VERSION="2023-11-01"
+AKS_ENDPOINT="https://${ARM_ENDPOINT}/providers/Microsoft.ContainerService/operations?api-version=${API_VERSION}"
+APISERVER_ENDPOINT="https://${APISERVER_FQDN}/healthz"
+ACS_BINARY_ENDPOINT="acs-mirror.azureedge.net/azure-cni/v1.4.43/binaries/azure-vnet-cni-linux-amd64-v1.4.43.tgz"
```

Comment:
again, this particular CNI version is controlled by a toggle in RP, if that toggle is updated, whats the plan to update it here?

Reply:
The reason why I used a specific endpoint is to get a 200 status code when using curl because this is a repository endpoint so just doing a generic path like this `curl https://acs-mirror.azureedge.net` will return a 400 status code saying bad request error. However, since the purpose of this test is to check if connection is blocked by any firewall/nsg, 400 is a good indication that there's connection so I guess we should be fine with 400 status code. Thus, I updated the endpoint to remove the specific version, so we don't have to deal with updating version constantly or if the binary is taken down.